File: Hibernate/src/main/java/grpc/converter/userProto/ConverterUser.java
package grpc.converter.userProto;

import GRPC.proto.File.*;
import domain.Model.User;

public class ConverterUser
{
  public static UserProto User_To_UserProto(User user)
  {
    return UserProto.newBuilder()
        .setId(user.getId())
        .build();
  }


  public static User UserProto_To_User(UserProto userProto)
  {
    return new User(userProto.getId());
  }

  /*
  public static User CreateUserRequest_To_User(CreateUserRequest newUser){
    return new User(CreateUserRequest.);
  }

   */
}

File: LogicClient/Converters/ConverterUser.cs
using GRPC.Proto;
using Shared.DTOs.User;
using Shared.Model;

namespace ClientgRPC.Converters;

public class ConverterUser
{
    
    public static UserProto UserToUserProto(User user)
    {
        return new UserProto {Id = user.Id};
    }

    public CreateUserRequest CreationToProto(UserCreationDto dto) {
        return new CreateUserRequest {
            IdUser = dto.id,
            Role = "temp"
        };
    }

    public static User ProtoToUser(UserProto proto) {
        return new User(proto.Id, proto.Role);
    }

    public UserSearchRequest SearchToProto(UserSearchDto dto) {
        return new UserSearchRequest {
            Id = dto.Id,
            Role = "temp"
        };
    }

    public List<User> ProtoToList(UserListProto proto) {
        List<User> returnList = new();

        foreach (var index in proto.List) {
            returnList.Add(ProtoToUser(index));
        }

        return returnList;
    }
    
}